sdk/search/azure-search/src/main/java/com/azure/search/SearchIndexClient.java

```diff
@@ -0,0 +1,354 @@
+     * @param context additional context that is passed through the Http pipeline during the service call
+     * @return response containing the document index result.
+     */
+    public Response<IndexDocumentsResult> mergeDocumentsWithResponse(Iterable<?> documents, Context context) {
+        return asyncClient.mergeDocumentsWithResponse(documents, context).block();
+    }
+
+    /**
+     * This action behaves like merge if a document with the given key already exists in the index.
+     * If the document does not exist, it behaves like upload with a new document.
+     *
+     * @param documents collection of documents to be merged, if exists, otherwise uploaded
+     * @return document index result
+     */
+    public IndexDocumentsResult mergeOrUploadDocuments(Iterable<?> documents) {
+        return this.mergeOrUploadDocumentsWithResponse(documents, Context.NONE).getValue();
+    }
+
+    /**
+     * This action behaves like merge if a document with the given key already exists in the index.
+     * If the document does not exist, it behaves like upload with a new document.
+     *
+     * @param documents collection of documents to be merged, if exists, otherwise uploaded
+     * @param context additional context that is passed through the Http pipeline during the service call
+     * @return response containing a document index result
+     */
+    public Response<IndexDocumentsResult> mergeOrUploadDocumentsWithResponse(Iterable<?> documents, Context context) {
+        return asyncClient.mergeOrUploadDocumentsWithResponse(documents, context).block();
+    }
+
+    /**
+     * Deletes a collection of documents from the target index
+     *
+     * @param documents collection of documents to delete from the target Index.
+     * @return document index result.
+     */
+    public IndexDocumentsResult deleteDocuments(Iterable<?> documents) {
+        return this.deleteDocumentsWithResponse(documents, Context.NONE).getValue();
+    }
+
+    /**
+     * Deletes a collection of documents from the target index
+     *
+     * @param documents collection of documents to delete from the target Index.
+     * @param context additional context that is passed through the Http pipeline during the service call
+     * @return response containing a document index result.
+     */
+    public Response<IndexDocumentsResult> deleteDocumentsWithResponse(Iterable<?> documents, Context context) {
+        return asyncClient.deleteDocumentsWithResponse(documents, context).block();
+    }
+
+    /**
+     * Gets Client Api Version.
+     *
+     * @return the apiVersion value.
+     */
+    public SearchServiceVersion getApiVersion() {
+        return asyncClient.getApiVersion();
+    }
+
+    /**
+     * Gets the endpoint for the Azure Cognitive Search service.
+     *
+     * @return the endpoint value.
+     */
+    public String getEndpoint() {
+        return this.asyncClient.getEndpoint();
+    }
+
+    /**
+     * Gets the number of documents
```

Comment:
@navalev The documentation here is not great compared to that in the .NET SDK. At least the method summaries should be copied over from this file:

https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/search/Microsoft.Azure.Search.Data/src/Customizations/Documents/IDocumentsOperations.Customization.cs

Some methods also have detailed remarks that explain the serialization/deserialization behavior for different data types. I'll point those out where they're missing.

Reply:
countDocument desc updated